pandora-common/src/gameLogic/characterModifiers/modifierTypes/speech_doll_talk.ts

```diff
@@ -27,16 +27,28 @@ _Note: Setting '0' means this part is not limited._
 				min: 0,
 			},
 		},
+		wordAllowlist: {
+			name: 'Words that can still be used',
+			type: 'stringList',
+			default: [],
+			options: {
+				maxCount: 100,
+				maxEntryLength: 24,
+				matchEntry: /^\p{L}+$/igu,
+			},
+		},
 	},
 	checkChatMessage(config, message) {
 		if (message.type !== 'chat')
 			return { result: 'ok' };
 
+		const allowedWordSet = new Set(config.wordAllowlist.map((w) => w.toLocaleLowerCase()));
```

Comment:
Please don't use `Locale*` methods in code that runs both on server and the client. While it is rare for the behavior to differ between locales, there certainly are cases where that happens - in that case the code would behave differently on client and server, which could lead to issues.
Use `toLowerCase` instead.